analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/DeliveringDebugFeaturesInProduction.cs

```diff
@@ -30,9 +30,15 @@ public sealed class DeliveringDebugFeaturesInProduction : DeliveringDebugFeature
         internal /*for testing*/ DeliveringDebugFeaturesInProduction(IAnalyzerConfiguration configuration) : base(configuration) { }
 
         // For simplicity we will avoid creating noise if the validation is invoked in the same method (https://github.com/SonarSource/sonar-dotnet/issues/5032)
-        protected override bool IsDevelopmentCheckInvoked(SyntaxNode node, SemanticModel semanticModel) =>
-            node.FirstAncestorOrSelf<MethodDeclarationSyntax>() is { } parentMethodDeclaration
-            && parentMethodDeclaration.DescendantNodes().Any(x => IsDevelopmentCheck(x, semanticModel));
+        protected override bool IsDevelopmentCheckInvoked(SyntaxNode node, SemanticModel semanticModel)
+        {
+            SyntaxNode parentMethod = node.FirstAncestorOrSelf<MethodDeclarationSyntax>();
```

Comment:
```suggestion
            SyntaxNode parentMethod = node.FirstAncestorOrSelf<BaseMethodDeclarationSyntax>();
```

Reply:
Misses local functions, lambdas, property getter/setter, indexer getter/setter, event adder/remover, field initializer. This should look more like this:
https://github.com/SonarSource/sonar-dotnet/blob/71a67baddd08c31c8c97365b7bb74c66b397e1b5/analyzers/src/SonarAnalyzer.CSharp/Rules/NonAsyncTaskShouldNotReturnNull.cs#L63-L77

and should be an extension method: `GetContainer` or `GetEnclosingScope`